motionplan/check.go

```diff
@@ -222,34 +225,35 @@ func checkPlanRelative(
 	})
 
 	lastArcEndPose := currentArcEndPose
+
 	// iterate through remaining plan and append remaining segments to check
 	for i := wayPointIdx + 1; i <= len(plan.Path())-1; i++ {
-		thisArcEndPoseTf, ok := plan.Path()[i][checkFrame.Name()]
-		if !ok {
-			return errors.New("check frame given not in plan Path map")
-		}
-		thisArcEndPoseInWorld, err := toWorld(thisArcEndPoseTf, plan.Trajectory()[i])
+		thisArcEndPoseInWorld, err := toWorld(zeroPosePIF, plan.Trajectory()[i])
 		if err != nil {
 			return err
 		}
 		thisArcEndPose := spatialmath.Compose(thisArcEndPoseInWorld.Pose(), errorState)
-		// Starting inputs for relative frames should be all-zero
 		startInputs := map[string][]referenceframe.Input{}
 		for k, v := range plan.Trajectory()[i] {
 			if k == checkFrame.Name() {
-				startInputs[k] = make([]referenceframe.Input, len(checkFrame.DoF()))
+				// Starting inputs for relative frames should be all-zero
+				correctedInputs := make([]referenceframe.Input, ptgDoFLen)
+				correctedInputs = append(correctedInputs, v[ptgDoFLen:]...)
```

Comment:
This piece right here is still weird to me; I know why we need it right now, but it feels off. It's kind of the last non-generic piece of this. Will think more if we can remove this.

Reply:
What are your thoughts on this idea:
I create an public interface perhaps in the referenceframe package called something like:
`GainSubsets` with one method on it called `Subsets([]referenceframe.Input) ([]referenceframe.Input, []referenceframe.Input)` or `subsets([]referenceframe.Input) [][]referenceframe.Input, []referenceframe.Input`

The method `Subsets` can then be implemented on the wrapperFrame.
We can type assert that checkFrame fulfills the `GainSubsets` interface. 
We can then run `Subsets` on the asserted type and gain inputs pertaining to the localization and execution frame.

Is your current train of thought on we can remove the need for this to exist or more along my line of thought?

Reply:
The `Subsets` is an interesting idea I think independently of the issue at hand. I like it, but I don't think it solves this particular problem.

The issue I see is that we're still breaking the Input abstraction a bit here, in the sense that we have to "know" how to set the inputs here. Worse, in this specific scenario we have to "know" that it is the the first `ptgDoFLen` inputs that actually correspond to the PTG frame.

In the interest of moving business logic out of `motionplan`, I wonder if this could instead be captured in the `Interpolate` method of the `wrapper` frame; that can "know" what the PTG frame is, "know" how it's being used, and adjust inputs accordingly.

Reply:
> In the interest of moving business logic out of motionplan, I wonder if this could instead be captured in the Interpolate method of the wrapper frame; that can "know" what the PTG frame is, "know" how it's being used, and adjust inputs accordingly.

ooo, I see what you mean - good idea!
I will describe my understanding of what you wrote here in more depth.
Let us first describe a generic plan:
```
Path: [ s, p1, p2, g, g ]
Trajectory: [ [0,0,0,0], [i1, a1, ds1, de1], 
                         [i2, a2, ds2, de2], 
                         [i3, a3, ds3, de3], [0,0,0,0] ]

```
As the current moment this is what the segments given to interpolation look like:
```
[ [startPosition: s, endPosition: s, startConfiguration: [ [0,0,0,0, s] ], endConfiguration: [ [0,0,0,0, s] ] , 
  [startPosition: s, endPosition: p1, startConfiguration: [ [0,0,0,0, s] ], endConfiguration: [ [i1, a1, ds1, de1, s] ] , 
  [startPosition: p1, endPosition: p2, startConfiguration: [ [0,0,0,0, p1] ], endConfiguration: [ [i2, a2, ds2, de2, p1] ] , 
  [startPosition: p2, endPosition: g, startConfiguration: [ [0,0,0,0, p2] ], endConfiguration: [ [i3, a3, ds3, de3, p2 ] ] , 
  [startPosition: g, endPosition: g, startConfiguration: [ [0,0,0,0, g] ], endConfiguration: [ [0,0,0,0, g] ] ]
```

In order to move "business logic out of motionplan" and have knowledge of how to handle this be instead captured by the `Interpolate` method of the `wrapperFrame`, we may actually change what the segments given to interpolation look like:

Now they may look like:
```
[ [startPosition: s, endPosition: s, startConfiguration: [ [0,0,0,0, s] ], endConfiguration: [ [0,0,0,0, s] ] , 
  [startPosition: s, endPosition: p1, startConfiguration: [ [0,0,0,0, s] ], endConfiguration: [ [i1, a1, ds1, de1, s] ] , 
  [startPosition: p1, endPosition: p2, startConfiguration: [ [i1, a1, ds1, de1, p1] ], endConfiguration: [ [i2, a2, ds2, de2, p1] ] , 
  [startPosition: p2, endPosition: g, startConfiguration: [ [i2, a2, ds2, de2, p2] ], endConfiguration: [ [i3, a3, ds3, de3, p2 ] ] , 
  [startPosition: g, endPosition: g, startConfiguration: [ [i3, a3, ds3, de3, p2, g] ], endConfiguration: [ [0,0,0,0, g] ] ]
```

Then interpolation of the ptgFrame may only care about the `endConfiguration` values and our `startConfiguration` will simply always be `make([]referenceframe.Input, len(wf.executionFrame.DoF()))`

Reply:
Exactly! We are on the same page.

`CheckPlan` need not then care about any specific frame types for segments; it simply constructs everything the same way.